document/js-api/index.bs

```diff
@@ -623,7 +634,21 @@ Immediately after a WebAssembly [=memory.grow=] instruction executes, perform th
 </div>
 
 <div algorithm>
-    The getter of the <dfn attribute for="Memory">buffer</dfn> attribute of {{Memory}} returns the receiver's \[[BufferObject]] internal slot.
+    The getter of the <dfn attribute for="Memory">buffer</dfn> attribute of {{Memory}} perfoms the following steps:
+
+    1. Let |memory| be the Memory instance.
+    1. Let |memaddr| be |memory|.\[[Memory]].
+    1. Let |block| be a [=Data Block=] which is [=identified with=] the underlying memory of |memaddr|.
+    1. If |block| is a [=Data Block=],
+        1. Return |memory|.\[[BufferObject]].
+    1. Otherwise,
+        1. Assert: |block| is a [=Shared Data Block=].
+        1. Let |map| be the [=surrounding agent=]'s associated [=Memory object cache=].
+        1. Assert: |map|[|memaddr|] [=map/exists=].
+        1. Let |newMemory| be |map|[|memaddr|].
```

Comment:
I don't understand what's going on here. It seems like `memory` and `newMemory` should always be the same object?

Reply:
It's possible to have two memory objects that are associated with the same data block, for example if a `Memory` object is postMessage'd to a worker and back. The one that grew was written to the cache, but the other is not.

Reply:
Do you have any Web API spec text drafted for how serialization works for Memory?

Reply:
Not yet, I'll work on that next.

Reply:
Oh, you already did this here: https://github.com/WebAssembly/threads/pull/90. Do we need to update this at all?

Reply:
Haha I forgot! Well, this change makes @Ms2ger 's question clear: It will actually be the *same* memory object! In particular, when the shared Memory object is postMessaged back, the original Memory object will be found in the cache.

Reply:
Oh, right. So this actually ends up being much simpler then.